### PR TITLE
Add dry run benchmark cell model for testing communication scaling

### PR DESCRIPTION
**Pull request by bcumming**
Add a modified version of the benchmark cell example that uses dry run scaling.

### CODE REVIEW

File: example/drybench/readme.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Dryrun Example
+
+A miniapp that demonstrates how to use dry-run mode to simulate the effect of communication scaling
+with benchmark cells. This is useful for evaluating the communication overheads as a model
+is weak-scaled over MPI.
+
+
```
Comment (thorstenhater) [inline example/drybench/readme.md:7]: This newline could be replaced by a more in-depth description. 
- Dry-run does not set up real MPI ranks, instead a single rank is used.
- The remaining ranks are fake and just generate spikes for the local tile to process.

**Review by thorstenhater — CHANGES_REQUESTED** [review]
Just a question on the Readme, rest looks fine.

File: example/drybench/readme.md (hunk 1 above)
Comment (bcumming) [reply example/drybench/readme.md:7]: You are right that "communication overheads" would include MPI communication, which this benchmark does not.
Instead, if allows us to focus on the overheads of spike and event handing as the number of MPI ranks increases, without having to use MPI.

Comment (thorstenhater) [reply example/drybench/readme.md:7]: Should I add a comment regarding this to the PR, or do you want to do that yourself. As it stands it's the only blocker.

Comment (thorstenhater) [reply example/drybench/readme.md:7]: @bcumming 👋

**Review by thorstenhater — APPROVED** [review]
Thanks 👍

**Comment by brenthuisman** [comment]
Tests are not passing however, problem?

**Comment by noraabiakar** [comment]
bors try

**Pull request merged**